src/storage/src/hummock/local_version.rs

```diff
@@ -64,8 +64,10 @@ pub enum SyncUncommittedDataStage {
     CheckpointEpochSealed(BTreeMap<HummockEpoch, SharedBuffer>),
     /// Task payload when we start syncing
     Syncing(OrderSortedUncommittedData),
+    /// Sync task is failed
+    Failed(OrderSortedUncommittedData),
```

Comment:
IIUC, in the current implementation, `Failed` is the final stage. In other words, after a task fails, it will be in the `Failed` stage forever. This can be an issue.

Reply:
After a sync epoch enter the `Failed` stage, any `sync` on it will return an `Err`, which will trigger a recovery and clear the shared buffer data and later recalculate the epoch. Therefore, the `Failed` stage will not be there forever. The fail-recovery logic is consistent with the current main code.

Reply:
I see. Looks good to me.